### ISSUE TITLE
FLAKE: client_lb_end2end_test - assertion failure

### ISSUE THREAD

**Issue by yashykt** [grpc/grpc#20319]
https://source.cloud.google.com/results/invocations/1dd40bc7-def4-41fc-bd1e-151615464036/targets/%2F%2Ftest%2Fcpp%2Fend2end:client_lb_end2end_test@poller%3Dpoll/log

seen on #20317

```
[ RUN      ] ClientLbInterceptTrailingMetadataTest.BackendMetricData
E0920 00:36:14.745302655      17 lb_policy_registry.cc:39]   assertion failed: strcmp(factories_[i]->name(), factory->name()) != 0

```

**Comment by veblush** [comment]
It seems you're already on this :)

**Comment by markdroth** [comment]
I believe this was fixed by #22801.
